### PR TITLE
Cleanup FINDTEXTW

### PR DESCRIPTION
**Pull request by hughbe**
## Proposed Changes
- Cleanup FINDTEXTW

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/dotnet/winforms/pull/2791)

### CODE REVIEW

File: src/System.Windows.Forms/src/System/Windows/Forms/RichTextBox.cs [hunk 1]
```diff
@@ -2093,20 +2088,23 @@ public int Find(string str, int start, int end, RichTextBoxFinds options)
...
                 // The default for RichEdit 2.0 is to search in reverse
-                findOptions |= RichTextBoxConstants.FR_DOWN;
+                findOptions |= Comdlg32.FR.DOWN;
             }
 
             // Perform the find, will return ubyte position
             int position;
 
-            position = (int)UnsafeNativeMethods.SendMessage(new HandleRef(this, Handle), RichEditMessages.EM_FINDTEXT, findOptions, ft);
+            fixed (char* pText = str)
```
Comment (RussKie) [inline src/System.Windows.Forms/src/System/Windows/Forms/RichTextBox.cs]: `pText` isn't used, look like you forgot to set it in `ft`

File: src/System.Windows.Forms/src/System/Windows/Forms/RichTextBox.cs [hunk 2]
```diff
@@ -2074,17 +2068,18 @@ public int Find(string str, int start, int end, RichTextBoxFinds options)
...
             if ((options & RichTextBoxFinds.MatchCase) == RichTextBoxFinds.MatchCase)
             {
-                findOptions |= RichTextBoxConstants.FR_MATCHCASE;
+                findOptions |= Comdlg32.FR.MATCHCASE;
             }
 
+            bool selectWord = true;
             if ((options & RichTextBoxFinds.NoHighlight) == RichTextBoxFinds.NoHighlight)
             {
                 selectWord = false;
```
Comment (RussKie) [inline src/System.Windows.Forms/src/System/Windows/Forms/RichTextBox.cs]:
```cs
bool selectWord = (options & RichTextBoxFinds.NoHighlight) != RichTextBoxFinds.NoHighlight;
```

File: src/System.Windows.Forms/src/System/Windows/Forms/RichTextBox.cs (hunk 1 above)
Comment (hughbe) [reply src/System.Windows.Forms/src/System/Windows/Forms/RichTextBox.cs]: should've marked this as wip sorry i need to add tests when other PRs (#2607) are merged

Comment (RussKie) [reply src/System.Windows.Forms/src/System/Windows/Forms/RichTextBox.cs]: I wondered whether if should be :) All good

Comment (hughbe) [reply src/System.Windows.Forms/src/System/Windows/Forms/RichTextBox.cs]: Now waiting on #2777 and #2800

**Comment by hughbe** [comment]
OK. Tests added and all good. There was one regression made during porting (ordering of checking of nullness of parameter `str`) which I have fixed

**Pull request merged**